dataflow/src/backlog/mod.rs

```diff
@@ -122,21 +122,36 @@ pub(crate) struct WriteHandle {
 
 type Key<'a> = Cow<'a, [DataType]>;
 pub(crate) struct MutWriteHandleEntry<'a> {
-    handle: &'a mut multiw::Handle,
+    handle: &'a mut WriteHandle,
     key: Key<'a>,
 }
 pub(crate) struct WriteHandleEntry<'a> {
-    handle: &'a multiw::Handle,
+    handle: &'a WriteHandle,
     key: Key<'a>,
 }
 
 impl<'a> MutWriteHandleEntry<'a> {
     pub fn mark_filled(self) {
-        self.handle.clear(self.key)
+        if let Some((None, _)) = self.handle
+            .handle
+            .meta_get_and(self.key.clone(), |rs| rs.is_empty())
```

Comment:
This clone is unfortunate, but I think required, as we need self.key later on line 139.

Reply:
I think you can just use `&*self.key` here?